components/org.wso2.carbon.identity.sso.saml/src/main/java/org/wso2/carbon/identity/sso/saml/builders/SignKeyDataHolder.java

```diff
@@ -144,8 +150,14 @@ public SignKeyDataHolder(String username) throws IdentityException {
                 }
             }
 
+        } catch (IdentityException e) {
+            throw new IdentityException("Unable to access realm service", e);
```

Comment:
# key alias is null or empty in this case
# tenantDomain is always super tenant domain